From: "B. Stack" <bgstack15@gmail.com>
Date: Sat, 15 Mar 2025 09:03:11 +0100
Subject: update compilation flags for Debian & derivatives

Co-authored-by: Fab Stz <fabstz-it@yahoo.fr>
Source: Original research
Last-Modified: 2022-08-03
Last-Version: 11.23
Forwarded: not-needed

   - CPPFLAGS: required because it is set in debian/rules
   - LDFLAGS: remove -s because stripping this is done in dh_strip. If we
     kept it, the dbgsym packages would be almost empty.
   - LDFLAGS: remove -no-pie because this would conflict with
     DEB_BUILD_MAINT_OPTIONS set in debian/rules. The value hardening=+all
     enables PIE, while -no-pie would conflit with that.
---
 FreeFileSync/Source/Makefile              | 4 ++--
 FreeFileSync/Source/RealTimeSync/Makefile | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/FreeFileSync/Source/Makefile b/FreeFileSync/Source/Makefile
index fc2576b..114b9fd 100644
--- a/FreeFileSync/Source/Makefile
+++ b/FreeFileSync/Source/Makefile
@@ -8,7 +8,7 @@ CXXFLAGS += -std=c++23 -pipe -DWXINTL_NO_GETTEXT_MACRO -I../.. -I../../zenXml -i
            -Wall -Wfatal-errors -Wmissing-include-dirs -Wswitch-enum -Wcast-align -Wnon-virtual-dtor -Wno-unused-function -Wshadow -Wno-maybe-uninitialized \
            -O3 -DNDEBUG `$(WX_CONFIG) --cxxflags --debug=no` -pthread
 
-LDFLAGS += -s `$(WX_CONFIG) --libs std, aui, richtext --debug=no` -pthread
+LDFLAGS += `$(WX_CONFIG) --libs std, aui, richtext --debug=no` -pthread
 
 
 CXXFLAGS  += `$(PKG_CONFIG) --cflags openssl`
@@ -131,7 +131,7 @@ all: ../Build/Bin/$(exeName)
 
 $(tmpPath)/ffs/src/%.o : %
 	mkdir -p $(dir $@)
-	$(CXX) $(CXXFLAGS) -c $< -o $@
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $< -o $@
 
 clean:
 	rm -rf $(tmpPath)
diff --git a/FreeFileSync/Source/RealTimeSync/Makefile b/FreeFileSync/Source/RealTimeSync/Makefile
index 69b8756..cb90781 100644
--- a/FreeFileSync/Source/RealTimeSync/Makefile
+++ b/FreeFileSync/Source/RealTimeSync/Makefile
@@ -8,7 +8,7 @@ CXXFLAGS += -std=c++23 -pipe -DWXINTL_NO_GETTEXT_MACRO -I../../.. -I../../../zen
            -Wall -Wfatal-errors -Wmissing-include-dirs -Wswitch-enum -Wcast-align -Wnon-virtual-dtor -Wno-unused-function -Wshadow -Wno-maybe-uninitialized \
            -O3 -DNDEBUG `$(WX_CONFIG) --cxxflags --debug=no` -pthread
 
-LDFLAGS += -s `$(WX_CONFIG) --libs std, aui, richtext --debug=no` -pthread
+LDFLAGS += `$(WX_CONFIG) --libs std, aui, richtext --debug=no` -pthread
 
 
 CXXFLAGS += `$(PKG_CONFIG) --cflags gtk+-3.0`
@@ -64,7 +64,7 @@ all: ../../Build/Bin/$(exeName)
 
 $(tmpPath)/ffs/src/rts/%.o : %
 	mkdir -p $(dir $@)
-	$(CXX) $(CXXFLAGS) -c $< -o $@
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $< -o $@
 
 clean:
 	rm -rf $(tmpPath)
